src/utils/pgwire/src/error.rs

```diff
@@ -22,43 +22,43 @@ pub type PsqlResult<T> = std::result::Result<T, PsqlError>;
 /// Error type used in pgwire crates.
 #[derive(Error, Debug)]
 pub enum PsqlError {
-    #[error("Startup Error when connect to session: {0}")]
+    #[error("Failed to start a new session: {0}")]
     StartupError(#[source] BoxedError),
 
     #[error("Invalid password")]
     PasswordError,
 
-    #[error("QueryError: {0}")]
-    QueryError(#[source] BoxedError),
+    #[error("Failed to run the query: {0}")]
+    SimpleQueryError(#[source] BoxedError),
 
-    #[error("ParseError: {0}")]
-    ParseError(#[source] BoxedError),
+    #[error("Failed to prepare the statement: {0}")]
+    ExtendedPrepareError(#[source] BoxedError),
 
-    #[error("ExecuteError: {0}")]
-    ExecuteError(#[source] BoxedError),
+    #[error("Failed to execute the statement: {0}")]
+    ExtendedExecuteError(#[source] BoxedError),
```

Comment:
Another idea could be to make all these variants `transparent` since they don't seem to provide much useful context.

This actually reminds me of the proposal from @fuyufjh that whether we should put the root cause (the bottom of the source chain) at the very first line of the error reports. In current practice, the first line acts like a summary of the error. It could be friendly, but in most cases users have to go through the `Caused by` sections to find something useful.

Reply:
For example, if they're made transparent, this will be confusing:

```
dev=> alter system set not_exist_key to value;
ERROR:  gRPC request to meta service failed: Internal error

Caused by:
  SystemParams error: unrecognized system param "not_exist_key"
```